dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcInvocation.java

```diff
@@ -85,23 +91,68 @@ public DecodeableRpcInvocation(FrameworkModel frameworkModel, Channel channel, R
         this.request = request;
         this.inputStream = is;
         this.serializationType = id;
-        this.callbackServiceCodecFactory = CacheableSupplier.newSupplier(()->
+        this.callbackServiceCodecFactory = CacheableSupplier.newSupplier(() ->
             new CallbackServiceCodec(frameworkModel));
     }
 
+
+    public ObjectInput getObjectInput() {
+        return objectInput;
+    }
+
+    public void setObjectInput(ObjectInput objectInput) {
+        this.objectInput = objectInput;
+    }
+
     @Override
     public void decode() throws Exception {
+        boolean finishDecode = false;
         if (!hasDecoded && channel != null && inputStream != null) {
             try {
                 decode(channel, inputStream);
+                finishDecode = true;
             } catch (Throwable e) {
-                if (log.isWarnEnabled()) {
+                if (log.isWarnEnabled() && !(e instanceof ServiceNotFoundException)) {
                     log.warn(PROTOCOL_FAILED_DECODE, "", "", "Decode rpc invocation failed: " + e.getMessage(), e);
                 }
                 request.setBroken(true);
-                request.setData(e);
+                request.setError(e);
             } finally {
                 hasDecoded = true;
+                if (finishDecode) {
+                    ObjectInput in = getObjectInput();
+                    if (in instanceof Cleanable) {
+                        ((Cleanable) in).cleanup();
+                    }
+                }
+                // Whether the process ends normally or abnormally (retry at most once), resources must be released
+            }
+        }
+    }
+
+    @Override
+    public void retry() {
+
+        String exPs = ApplicationModel.defaultModel().getCurrentConfig().getParameters().get(EXCEPTION_PROCESSOR_KEY);
```

Comment:
`DecodeableRpcInvocation` here is a framework level class, and cannot get application model here.